### PR TITLE
Feature bigtable

### PR DESCRIPTION
**Pull request by NavdeepCS**

### CODE REVIEW

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 1]
```diff
@@ -1296,4 +1320,113 @@ private static String createGCSBucketLifeCycle() throws IOException, URISyntaxEx
...
+      PluginPropertyUtils.removePluginProp("bigtableTargetCluster");
+      PluginPropertyUtils.removePluginProp("bigtableTargetExistingTable");
+      bigtableTargetInstance=StringUtils.EMPTY;
+      bigtableTargetCluster=StringUtils.EMPTY;
+      bigtableExistingTargetTable=StringUtils.EMPTY;
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+}
```
Comment (psainics) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: Add new line at end of files

File: src/e2e-test/java/io/cdap/plugin/bigtable/runners/TestRunner.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        features = {"src/e2e-test/features"},
+        glue = {"io.cdap.plugin.bigtable.stepsdesign", "io.cdap.plugin.common.stepsdesign", "stepsdesign"},
+        tags = {"@BigTable"},
+        monochrome = true,
+        plugin = {"pretty", "html:target/cucumber-html-report/bigtable",
+                "json:target/cucumber-reports/cucumber-bigtable.json",
+                "junit:target/cucumber-reports/cucumber-bigtable.xml"}
+)
+public class TestRunner {
+}
```
Comment (psainics) [inline src/e2e-test/java/io/cdap/plugin/bigtable/runners/TestRunner.java]: Add new line at end of files

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 3]
```diff
@@ -69,6 +77,22 @@ public class TestSetupHooks {
...
+
+  public static String bigtableSourceTable = StringUtils.EMPTY;
+  public static String bigtableTargetInstance = StringUtils.EMPTY;
+  public static String bigtableTargetCluster = StringUtils.EMPTY;
+
+  public static String bigtableTargetTable = StringUtils.EMPTY;
+  public static String bigtableExistingTargetTable = StringUtils.EMPTY;
+
+  public static BigtableInstanceAdminSettings instanceAdminSettings;
+  public static  BigtableInstanceAdminClient adminClient;
```
Comment (psainics) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: nit: extra space

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 1 above)
Comment (NavdeepCS) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: Fixed,thanks.

File: src/e2e-test/java/io/cdap/plugin/bigtable/runners/TestRunner.java (hunk 2 above)
Comment (NavdeepCS) [reply src/e2e-test/java/io/cdap/plugin/bigtable/runners/TestRunner.java]: Fixed,thanks.

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 3 above)
Comment (NavdeepCS) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: Fixed,thanks.

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 4]
```diff
@@ -1296,4 +1310,109 @@ private static String createGCSBucketLifeCycle() throws IOException, URISyntaxEx
   public static void createBucketWithLifeCycle() throws IOException, URISyntaxException {
     gcsTargetBucketName = createGCSBucketLifeCycle();
     BeforeActions.scenario.write("GCS target bucket name - " + gcsTargetBucketName); }
+
+
```
Comment (itsmekumari) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:1381]: Remove extra line

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 3 above)
Comment (psainics) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: There is an extra space between `static` and `BigtableInstanceAdminClient`

File: src/e2e-test/java/io/cdap/plugin/utils/BigTableClient.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.client.Table;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.hadoop.hbase.client.Get;
+import org.codehaus.commons.nullanalysis.NotNull;
+import org.apache.hadoop.hbase.client.Result;
+import org.junit.Assert;
+
+
```
Comment (itsmekumari) [inline src/e2e-test/java/io/cdap/plugin/utils/BigTableClient.java:24]: Remove extra line

File: src/e2e-test/bigtable/BigTableToBigTable.feature [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    Then Close the pipeline logs
+    Then Close the preview
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Enter runtime argument value "cbtSourceOutputSchema" for key "cbtSourceOutputSchema"
+    Then Run the Pipeline in Runtime with runtime arguments
+    Then Wait till pipeline is in running state
+    Then Verify the pipeline status is "Succeeded"
+    Then Validate OUT record count is equal to IN record count
+
```
Comment (itsmekumari) [inline src/e2e-test/bigtable/BigTableToBigTable.feature]: Add a data validation step for scenarios

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 3 above)
Comment (NavdeepCS) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: Fixed,thanks

**Pull request closed without merging**